source/GameInterface/Utils/GenericCollectionPatches.cs

```diff
@@ -401,29 +532,60 @@ public static void ArrayAssignIntercept<TItem, TMessage>(TItem[] items, int inde
         }
 
-        private static IEnumerable<CodeInstruction> PatchPropertyInstructions(List<CodeInstruction> instructionList, List<CodeInstruction> propertyInstructions, MethodInfo addMethod, MethodInfo removeMethod, MethodInfo addIntercept, MethodInfo removeIntercept)
+        private static IEnumerable<CodeInstruction> PatchInstructions(List<CodeInstruction> instructionList, 
+            Func<CodeInstruction, bool> targetLocator,
+            MethodInfo addMethod, 
+            MethodInfo addIntercept, 
+            MethodInfo removeMethod, 
+            MethodInfo removeIntercept
+            )
         {
-            foreach (var propInst in propertyInstructions)
+            var targetInstructions = instructionList.Where(ci => targetLocator(ci)).ToList();
+
+            foreach (var targetInst in targetInstructions)
             {
-                int propIndex = instructionList.IndexOf(propInst);
-                var range = instructionList.GetRange(propIndex + 1, 2);
+                int targetIndex = instructionList.IndexOf(targetInst);
+
+                // Check if we have enough instructions left so that there could be a call to the Add/Remove-Method
+                // Usually there is an ldarg and then the callvirt
+                if (instructionList.Count < targetIndex + 3)
+                    return instructionList;
+
+                // Check the next 2 instructions for the Callvirt
+                var range = instructionList.GetRange(targetIndex + 1, 2);
 
                 var changeInst = range.FirstOrDefault(inst => inst.opcode == OpCodes.Callvirt && (inst.operand as MethodInfo == addMethod || inst.operand as MethodInfo == removeMethod));
+                // Add the instance that contains our target right before the Callvirt instruction
                 if (changeInst != null)
                 {
                     int changeIndex = instructionList.IndexOf(changeInst);
-
-                    var prevPropInst = instructionList[propIndex - 1];
-                    instructionList.RemoveRange(changeIndex - 1, 1);
-
-                    changeIndex = instructionList.IndexOf(changeInst);
-                    instructionList.Insert(changeIndex, new CodeInstruction(OpCodes.Ldarg_0));
-                    changeIndex = instructionList.IndexOf(changeInst);
-                    instructionList.InsertRange(changeIndex, instructionList.GetRange(propIndex - 2, 2));
+                    // Search beginning of loadStack to be able to retrieve the instance that the collection belongs to
+                    var stack = new List<CodeInstruction>();
+                    for(int i = targetIndex-1; i >=0; i--)
+                    {
+                        var currentInst = instructionList[i];
+                        stack.Add(currentInst);
+                        if (IsLdArg(currentInst))
+                            break;
+                    }
+                    stack.Reverse();
```

Comment:
Wouldnt adding to the stack do the same thing?

if (IsLdArg(currentInst)) stack.add(instructionList[i]);